tools/audio/samplebank_compiler.c

```diff
@@ -0,0 +1,202 @@
+            }
+            arg_error("Unknown option \"%s\"", argv[i]);
+        } else {
+            // Required args
+
+            switch (argn) {
+                case 0:
+                    filename = argv[i];
+                    break;
+                case 1:
+                    outfilename = argv[i];
+                    break;
+                default:
+                    arg_error("Unknown positional argument \"%s\"", argv[i]);
+                    break;
+            }
+            argn++;
+        }
+    }
+    if (argn != 2)
+        arg_error("Not enough positional arguments");
+
+#undef arg_error
+
+    // open xml
+    document = xmlReadFile(filename, NULL, XML_PARSE_NONET);
+    if (document == NULL)
+        return EXIT_FAILURE;
+
+    // parse xml
+    read_samplebank_xml(&sb, document);
+
+    // open output asm file
+    outf = fopen(outfilename, "w");
+    if (outf == NULL)
+        error("Unable to open output file [%s] for writing", outfilename);
+
+    // open output dep file if applicable
+    if (mdfilename != NULL) {
+        mdfile = fopen(mdfilename, "w");
+        if (mdfile == NULL)
+            error("Unable to open dependency file [%s] for writing", mdfilename);
+
+        fprintf(mdfile, "%s: \\\n    %s", outfilename, filename);
+    }
+
+    // write output
+
+    fprintf(outf,
+            // clang-format off
+           ".rdata"                 "\n"
+           ".balign 16"             "\n"
+                                    "\n"
+           ".global %s_Start"       "\n"
+           "%s_Start:"              "\n"
+           "$start:"                "\n",
+            // clang-format on
+            sb.name, sb.name);
+
+    // original tool appears to have a buffer clearing bug involving a buffer sized BUG_BUF_SIZE
+    match_buf_ptr = (matching) ? match_buf : NULL;
+    match_buf_pos = 0;
+
+    for (size_t i = 0; i < sb.num_samples; i++) {
+        const char *name = sb.sample_names[i];
+        const char *path = sb.sample_paths[i];
+        bool is_sample = sb.is_sample[i];
+
+        if (mdfilename != NULL)
+            fprintf(mdfile, " \\\n    %s", path);
+
+        if (!is_sample) {
+            // blob
+            fprintf(outf,
+                    // clang-format off
+                                        "\n"
+                   "# BLOB %s"          "\n"
+                                        "\n"
+                   ".incbin \"%s\""     "\n"
+                                        "\n"
+                   ".balign 16"         "\n"
+                                        "\n",
+                    // clang-format on
+                    name, path);
+            continue;
+        }
+
+        // aifc sample
+        fprintf(outf,
+                // clang-format off
+                                                "\n"
+               "# SAMPLE %lu"                   "\n"
+                                                "\n"
+               ".global %s_%s_Abs"              "\n"
```

Comment:
this is pretty nitpicky but I think the name `SampleBank_0_SAMPLE_0_19_Abs` could be better:
* Different casing between SampleBank and SAMPLE
* Sample bank number is repeated twice

perhaps it should use the same naming as the .aifc files (including the leading zeros), like`SampleBank_0_Sample019_Abs`. Not a big deal though if you think this ship has sailed